src/modules/twap/updaters/TwapOrdersUpdater.tsx

```diff
@@ -0,0 +1,72 @@
+import { useUpdateAtom } from 'jotai/utils'
+import { useEffect, useMemo } from 'react'
+
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+
+import { ComposableCoW } from 'abis/types'
+
+import { useDiscreteOrdersFromOrderBook } from '../hooks/useDiscreteOrdersFromOrderBook'
+import { useFetchDiscreteOrders } from '../hooks/useFetchDiscreteOrders'
+import { useFetchTwapOrdersFromSafe } from '../hooks/useFetchTwapOrdersFromSafe'
+import { useTwapOrdersAuthMulticall } from '../hooks/useTwapOrdersAuthMulticall'
+import { twapDiscreteOrdersAtom } from '../state/twapDiscreteOrdersAtom'
+import { twapOrdersListAtom } from '../state/twapOrdersListAtom'
+import { TwapOrderInfo } from '../types'
+import { getConditionalOrderId } from '../utils/getConditionalOrderId'
+import { getTwapOrdersItems } from '../utils/getTwapOrdersItems'
+import { isTwapOrderExpired } from '../utils/getTwapOrderStatus'
+import { parseTwapOrderStruct } from '../utils/parseTwapOrderStruct'
+
+export function TwapOrdersUpdater(props: {
+  safeAddress: string
+  chainId: SupportedChainId
+  composableCowContract: ComposableCoW
+}) {
+  const { safeAddress, chainId, composableCowContract } = props
+
+  const discreteOrdersFromOrderBook = useDiscreteOrdersFromOrderBook()
+  const setTwapOrders = useUpdateAtom(twapOrdersListAtom)
+  const updateTwapDiscreteOrders = useUpdateAtom(twapDiscreteOrdersAtom)
+  const ordersSafeData = useFetchTwapOrdersFromSafe(props)
+
+  const allOrdersInfo: TwapOrderInfo[] = useMemo(() => {
+    return ordersSafeData.map((data) => {
+      const id = getConditionalOrderId(data.params)
+      const order = parseTwapOrderStruct(data.params.staticInput)
+
+      return {
+        id,
+        orderStruct: order,
+        safeData: data,
+        isExpired: isTwapOrderExpired(order),
+      }
+    })
+  }, [ordersSafeData])
+
+  // Here we can split all orders in two groups: 1. Not signed + expired, 2. Open + cancelled
+  const openOrCancelledOrders = useMemo(() => {
+    return allOrdersInfo.filter((info) => !info.isExpired && info.safeData.isExecuted)
+  }, [allOrdersInfo])
+
+  const discreteOrders = useFetchDiscreteOrders(safeAddress, chainId, composableCowContract, openOrCancelledOrders)
```

Comment:
I'm a bit confused with `discreteOrdersFromOrderBook` vs `discreteOrders`

Sounds like they should be the same, but I'm not sure they are

Reply:
I guess one is from the state, and the other is what composibeCowContract returns. Since this is the updater, it needs both to update the state. 

However, i agree is confusing. We should make sure the names of the vars reflect this.

Reply:
It's more than confusing, I agree.
Thanks @anxolin for the suggestion! I renamed `discrete` to `particle`:
https://github.com/cowprotocol/cowswap/pull/2597/commits/4a64868dfbb29a1cee2cb34023a2d165ed63e9c3

`particle` - it's an order from `ComposableCoW.getTradeableOrderWithSignature`
`discrete` - an order from order-book

Reply:
We should be careful calling discrete for orders from the order-book, these are actually orders from the protocol, and the most common entity.

If we call them `discrete` , all our hooks, state, etc will start to have `discrete` in their name, instead of being just `usePendingOrders` would be `usePrendingDiscreteOrders` which will make all our functions too verbose

It feels easier to add a suffix to the weird orders that are out of the protocol, like the twap orders.